MM-54278 - Calls banners redesign

cpoile opened the pull request:
#### Summary
- This PR changes and does a refactor of the calls banner design and UI, partly for polish, partly to accommodate the upcoming bookmarks feature.
- It also refactors the internal design (to rely more on flex gap instead of manually computing distances and placing banners with absolute positions). This simplifies a lot of the code.
- The voice-on (talking) shadow has been replaced by the dual border effect for iOS. This makes it look like Android now.
- The talking borders are a bit smaller in the current call, a bit bigger in the call screen.
- Emojis were also inconsistent across devices and avatar sizes, made them similar. Also tightened up some of the spacing between the avatars and the names
- Todo:
  - adjust the New Messages bar for the new UI, and place it under all the banners (after adding the error bars and the incoming call notifications it was hidden).
  - call from another server changes
  - calls banners on the channel list screen

#### Designs
- https://www.figma.com/file/WVbRdxA4aRFdw410j2DxxU/MM-51513-Ringing-in-DMs%2C-GMs-(Mobile)?type=design&node-id=1378-95706&mode=design&t=CoxDrNZEd8wk9Y0B-0

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-53956

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file updates
- [x] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
- Android: 13, Pixel 6
- iOS: 16.5.1, iPhone 14

#### Screenshots

Android: (ios looks the same)

NOTE: the first two call quality banners have the correct styling, imagine the others have it too.

|col1|col2|col3|
|---|----|---|
| ![Screenshot_20231017-174950](https://github.com/mattermost/mattermost-mobile/assets/1490756/3fd69a74-af74-48e9-8482-46ec49717dce) | ![Screenshot_20231017-172024](https://github.com/mattermost/mattermost-mobile/assets/1490756/15ec2223-964a-4392-8744-a635d37259b9) | ![Screenshot_20231017-172032](https://github.com/mattermost/mattermost-mobile/assets/1490756/969df6cb-a59e-424e-b238-119f3246f96a) |
| ![Screenshot_20231017-172055](https://github.com/mattermost/mattermost-mobile/assets/1490756/482bcb5a-66be-4e80-8c9a-197418f854e6) | ![Screenshot_20231017-175132](https://github.com/mattermost/mattermost-mobile/assets/1490756/431a95b8-0190-46c9-971a-98f0f1eb59f4) | ![Screenshot_20231017-171547](https://github.com/mattermost/mattermost-mobile/assets/1490756/3b1cbf58-2934-4e57-8c16-f9fe799421ff) |
| ![Screenshot_20231017-171431](https://github.com/mattermost/mattermost-mobile/assets/1490756/9ab228b5-1393-4498-8755-ab63e4af8a3e) | ![Screenshot_20231017-171409](https://github.com/mattermost/mattermost-mobile/assets/1490756/90118895-8772-405d-b70a-66aadb715bad) | ![Screenshot_20231017-171339](https://github.com/mattermost/mattermost-mobile/assets/1490756/1d1701e6-f88b-4c08-8e27-fca86ce65397) |
| ![Screenshot_20231017-165254](https://github.com/mattermost/mattermost-mobile/assets/1490756/ee38058b-4c9b-40bc-b72d-68609473eefd) | |

iOS so you can see that the emojis are a bit bigger now (matches Android), and the voice indicator is the same as Android.

| col1 | col2 |
|---|---|
| ![IMG_D899620474FB-1](https://github.com/mattermost/mattermost-mobile/assets/1490756/396af410-856b-4861-9bd1-00442dc1b5bb) |  ![IMG_85E8F01037AD-1](https://github.com/mattermost/mattermost-mobile/assets/1490756/c3de0208-19d4-430e-beec-210111228e76) |

#### Release Note

```release-note
Calls: UI redesign and polish
```

mattermost-build commented:
Building app in separate branch.

matthewbirtch reviewed (changes requested):
Great work @cpoile. You were very faithful to the designs :) I don't have much to comment on - just these few things. 

### All banners
- Are we able to add a shadow to the banners? Or is that not easy to pull off? Designs have a subtle shadow for all banners
- Also, I'd like to remove the borders on the banners - with the exception of the active call widget (that one needs it for dark themes). However, we can reduce the opacity on the border to 0.08.

### Active Call Widget
- For the 'no one is talking' state, I added a [screen in Figma](https://www.figma.com/file/WVbRdxA4aRFdw410j2DxxU/MM-51513-Ringing-in-DMs%2C-GMs-(Mobile)?type=design&node-id=1474-5588&mode=design&t=xHCELhvX8C16zlNx-4) that improves this a little. Could you update that while we're making changes here?
      ![image](https://github.com/mattermost/mattermost-mobile/assets/2040554/478794f3-1455-4da3-8f29-954be7c37fe8)

### Incoming call banner
- As discussed in our DM, I seem to be getting an in-app notification overlaid on top of the incoming call
    ![image](https://github.com/mattermost/mattermost-mobile/assets/2040554/cf5f011e-f31e-4767-8d2b-59d5304b803e)

### Call quality banner
- Looks like the `x` on the call quality banner has to update its color
    ![image](https://github.com/mattermost/mattermost-mobile/assets/2040554/b68d7465-0341-442d-af7d-2aa2a623e17c)
- Might need a _very_ slight adjustment for the icon and text to align with other banners. Or do my eyes deceive me?
    ![image](https://github.com/mattermost/mattermost-mobile/assets/2040554/bd198db3-c5e0-4c7b-ab1b-d5321fdd79bb)
- Same goes for the red microphone banner

cpoile commented:
@matthewbirtch Updated:
- light shadows (matching what we have for the New Messages banner, but I also added some for Android
- removed borders (except for Current call, and notifications in the Call Screen)
- made the changes for the 'no one is talking' state
- the push notification needs to be investigated: https://mattermost.atlassian.net/browse/MM-55036
- the banners should align now

| | |
|--|--|
| ![IMG_F966FD40DFA4-1](https://github.com/mattermost/mattermost-mobile/assets/1490756/c4acc868-d7ad-4414-b213-a34461a78149) | ![IMG_A76FBDE68F8F-1](https://github.com/mattermost/mattermost-mobile/assets/1490756/610b29a2-abde-4151-9518-2be4b7b5ff66) |
|  ![IMG_1326](https://github.com/mattermost/mattermost-mobile/assets/1490756/aa4b8efc-ddf3-43df-90d7-68da93502e93) |  ![IMG_1324](https://github.com/mattermost/mattermost-mobile/assets/1490756/389510fc-0f09-4680-b208-a7338965f277) |

matthewbirtch commented:
> @matthewbirtch Updated:
> 
> * light shadows (matching what we have for the New Messages banner, but I also added some for Android
> * removed borders (except for Current call, and notifications in the Call Screen)
> * made the changes for the 'no one is talking' state
> * the push notification needs to be investigated: https://mattermost.atlassian.net/browse/MM-55036
> * the banners should align now

Looks good @cpoile. Thanks for making those adjustments.

For the notification issue, I wanted to be clear that it's not actually a push notification that I'm seeing. It's an in-app notification. Those are used when you get notifications from other channels or DMs you're not currently viewing. So, for incoming calls, we should prevent those from showing. 

One other thing I just noticed in your screenshots is that the active call banner looks like it has too much padding on the left side now.

<img width="937" alt="image" src="https://github.com/mattermost/mattermost-mobile/assets/2040554/9022d253-5dec-4161-90f5-27510e206423">

enahum commented:
can you share a screenshot of the user avatar stack on the post list in a channel to see the impact on CRT posts ?

cpoile commented:
@matthewbirtch Good eye, here's the fix (I think it matches; I'm going by the figma distances and using those as the margin/paddings).

| | |
|--|--|
| ![IMG_1329](https://github.com/mattermost/mattermost-mobile/assets/1490756/f4ce2294-0fbb-46b8-88bd-695b4e1f09f9) | ![IMG_1330](https://github.com/mattermost/mattermost-mobile/assets/1490756/dcc43871-2aa5-490c-a21a-9b1b3550d267) |

@enahum Here's the pre (main) vs post (this pr):

| pre | post |
|--|--|
| ![IMG_FF963F2D6D39-1](https://github.com/mattermost/mattermost-mobile/assets/1490756/da32ec84-35ad-4849-85f8-c87be3c2bc06) | ![IMG_608042883F26-1](https://github.com/mattermost/mattermost-mobile/assets/1490756/1ae67597-942a-44c9-a1a4-0a0c1f326d45) |

matthewbirtch commented:
> @matthewbirtch Good eye, here's the fix (I think it matches; I'm going by the figma distances and using those as the margin/paddings).
> 
> ![IMG_1329](https://user-images.githubusercontent.com/1490756/276644231-f4ce2294-0fbb-46b8-88bd-695b4e1f09f9.jpg)	![IMG_1330](https://user-images.githubusercontent.com/1490756/276644285-dcc43871-2aa5-490c-a21a-9b1b3550d267.jpg)

👍 looks good now @cpoile. Were you going to look into the in-app notifications issue separately?

enahum commented on app/products/calls/components/incoming_calls_container.tsx:
```diff
@@ -3,63 +3,43 @@
-    const micPermissionsGranted = useGlobalCallsState().micPermissionsGranted;
-    const currentCall = useCurrentCall();
+
+    // const micPermissionsGranted = useGlobalCallsState().micPermissionsGranted;
+    // const currentCall = useCurrentCall();
```
on purpose?

cpoile replied:
nope, by accident, thanks!

enahum commented on app/products/calls/components/incoming_calls_container.tsx:
```diff
@@ -3,63 +3,43 @@
-        (micPermissionsError ? CALL_ERROR_BAR_HEIGHT + 8 : 0) +
-        (qualityAlert ? CALL_ERROR_BAR_HEIGHT + 8 : 0);
-    const wrapperTop = {top};
+    // const micPermissionsError = !micPermissionsGranted && (currentCall ? !currentCall.micPermissionsErrorDismissed : false);
+    // const qualityAlert = (currentCall ? currentCall.callQualityAlert && currentCall.callQualityAlertDismissed === 0 : false);
+    // const marginTop = (micPermissionsError ? CALL_ERROR_BAR_HEIGHT + 8 : 0) +
+    //     (qualityAlert ? CALL_ERROR_BAR_HEIGHT + 8 : 0);
+    // const wrapperTop = {marginTop};
```
on purpose?

cpoile replied:
same, thanks

streamer45 reviewed (approved):
Very thorough, thanks :+1:
